include/zephyr/dsp/dsp.h

```diff
@@ -17,6 +17,20 @@
 #define DSP_FUNC_SCOPE
 #endif
 
+#ifdef CONFIG_DSP_BACKEND_ARC
+#define AGU __agu
+#define DATA_ATTR __attribute__((section(".Xdata")))
+#else
+#define AGU
+#define DATA_ATTR
+#endif
+
+#ifdef CONFIG_DSP_POINTER_HAS_RESTRICT
+#define RESTRICT __restrict
+#else
+#define RESTRICT
+#endif
```

Comment:
I don't believe that we have a resolution on https://github.com/zephyrproject-rtos/zephyr/discussions/52617 yet. Can this PR be done without `__restrict` for now until we can discuss further?